pkg/csi/cinder/controllerserver.go

```diff
@@ -147,6 +167,38 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	return getCreateVolumeResponse(vol, ignoreVolumeAZ, req.GetAccessibilityRequirements()), nil
 }
 
+func GetSnapshot(cloud openstack.IOpenStack, snapshotID string) (error, error) {
+	snap, errsnap := cloud.GetSnapshotByID(snapshotID)
```

Comment:
Could you just use `err` instead of `errsnap`. This is not the case, when you need to differentiate between multiple `err` variables. The same applies to the `GetBackup` func.